File: src/entities/backlog/backlog.py
from src.entities.order.order import Order

class Backlog:
    """
    Representa o backlog de pedidos do galpão.

    Attributos:
        orders (list[Order]): lista de pedidos no backlog.
    """

    def __init__(self, orders: list[Order]) -> None:
        self.orders = orders

    def get_orders(self) -> list[Order]:
        return self.orders

    def get_order(self, order_id: int) -> Order:
        for order in self.orders:
            if order.order_id == order_id:
                return order

File: src/backlog/backlog.go
package backlog

import (
	"challenge_sbpo_2025/order"
)

type Backlog struct {
	orders []order.Order
}

func NewBacklog() *Backlog {
	return &Backlog{}
}

func (backlog *Backlog) AddOrder(order order.Order) {
	backlog.orders = append(backlog.orders, order)
}

func (backlog *Backlog) GetOrders() []order.Order {
	return backlog.orders
}
